src/lesson/router.ts

```diff
@@ -11,87 +11,188 @@ import * as handlers from './handlers';
-     *       201:
-     *         description: Lesson successfully created
-     *         content:
-     *           application/json:
-     *             schema:
-     *               $ref: '#/components/schemas/Lesson'
-     *       400:
-     *         description: Invalid video URL
-     *       500:
-     *         description: Server error during lesson creation
-     */
-    router.post('/', controller.createLesson);
+  /**
+   * @swagger
+   * /lesson:
+   *   post:
+   *     summary: Create a lesson from a video URL
+   *     tags: [Lesson]
+   *     requestBody:
+   *       required: true
+   *       content:
+   *         application/json:
+   *           schema:
+   *             type: object
+   *             required:
+   *               - videoUrl
+   *             properties:
+   *               videoUrl:
+   *                 type: string
+   *                 format: uri
+   *                 description: The URL of the video to summarize into a lesson
+   *           example:
+   *             videoUrl: "https://example.com/video.mp4"
+   *     responses:
+   *       201:
+   *         description: Lesson successfully created
+   *         content:
+   *           application/json:
+   *             schema:
+   *               $ref: '#/components/schemas/Lesson'
+   *       400:
+   *         description: Invalid video URL
+   *       500:
+   *         description: Server error during lesson creation
+   */
+  router.post("/", controller.createLesson);
 
-    return router;
+  /**
+   * @swagger
+   * /lesson:
+   *   get:
+   *     summary: Get all lessons
+   *     tags: [Lesson]
+   *     responses:
+   *       200:
+   *         description: A list of lessons
+   *         content:
+   *           application/json:
+   *             schema:
+   *               type: array
+   *               items:
+   *                 $ref: '#/components/schemas/Lesson'
+   *       500:
+   *         description: Server error
+   */
+  router.get("/", controller.getLessons);
+
+  /**
+   * @swagger
+   * /lesson/delete/{id}:
+   *   delete:
+   *     summary: Delete a lesson by its ID
+   *     tags: [Lesson]
+   *     parameters:
+   *       - in: path
+   *         name: id
+   *         required: true
+   *         schema:
+   *           type: string
+   *         description: ID of the lesson to delete
+   *     responses:
+   *       200:
+   *         description: Lesson successfully deleted
+   *         content:
+   *           application/json:
+   *             schema:
+   *               type: object
+   *               properties:
+   *                 message:
+   *                   type: string
+   *                   example: Lesson with id 123 deleted successfully.
+   *       404:
+   *         description: Lesson not found
+   *       400:
+   *         description: Invalid ID format
+   *       500:
+   *         description: Server error
+   */
+  router.delete("/delete/:id", controller.deleteLesson);
```

Comment:
no need for delete prefix, the method already says its delete
```suggestion
  router.delete("/:id", controller.deleteLesson);
```